packages/sanity/src/core/form/types/inputProps.ts

```diff
@@ -226,7 +234,7 @@ export interface ArrayOfObjectsInputProps<
   /**
    * @hidden
    * @beta */
-  onPathFocus: (path: Path) => void
+  onPathFocus: (path: Path, payload?: OnPathFocusPayload) => void
```

Comment:
Instead of adding a second argument, would it be more future proof to introduce a separate event type? I.e. change the signature of `onPathFocus(path: Path)` to `onPathFocus(event: InputFocusEvent | Path)`.

It would align with `PatchEvent` and give us an opportunity to add more properties to the event type later (instead of adding yet more positional arguments). If we did, we could deprecate the `onPathFocus(path: Path)` signature.

Reply:
@bjoerge - I think that sounds like a good idea. And we could individually tag it as @beta or whatever.